### PR TITLE
Touch calibration screen

### PR DESCRIPTION
**Pull request by rhapsodyv**
### Description

This PR add a touch calibration screen for all 3 marlin TFT UI.

Todo List:
 - [x] Class to control touch calibration logic
 - [x] Classic UI support
 - [x] Color UI support
 - [x] LVGL UI support 
 - [x] Remove duplicated `XPT2046_**` defines for the new `TOUCH_**`
 - [x] Remove XPT configuration from the board pins when the board uses SPI display
 - [x] Autostart touch calibration screen if no calibration values are saved on eeprom
 - [x] i18n

This PR closes the series for TFT changes I have done. TFT is not perfect yet, but now it will be fully usable and configurable by the users.

### Benefits

Without a calibration screen is nearly impossible to get the xpt calibration values. It's a lot of trial and error.

### Configurations

Any TFT UI with TOUCH_SCREEN_CALIBRATION

ezgif-6-91be9e3d2971

### CODE REVIEW

**Comment by rhapsodyv** [comment]
@thinkyhead  and @thisiskeithb 
I think it's ready for review. One thing is missing, but it needs discussion: should we remove xpt configs from pins file?

Maybe we should start automatically touch calibration screen after boot, if there's no value for XPT calibration... What do you thing?

When touch calibration screen is enable, it uses value from EEPROM, being the touch calibration macros default values. When disabled, it use only the macros and don't save anything.

**Comment by thisiskeithb** [comment]
> should we remove xpt configs from pins file?

Sure!

> Maybe we should start automatically touch calibration screen after boot, if there's no value for XPT calibration... What do you thing?

This is how it's done in the standalone TFT world currently and I think would be a perfectly fine approach.

**Comment by rhapsodyv** [comment]
It's done. Now if calibration values are 0, it will start the calibration screen after the bootscreen. I remove all xpt values from boards that have multiples screens possibilities. But I kept it in some boards that are FSMC with a unique and specific TFT. We can remove, but I don't think we need.

File: Marlin/src/lcd/extui/lib/mks_ui/draw_touch_calibration.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      default: break;
+    }
+
+    x = touch_calibration.calibration_points[calibration_stage].x;
+    y = touch_calibration.calibration_points[calibration_stage].y;
+    drawCross(x, y, LV_COLOR_WHITE.full);
+  }
+  else {
+    // end calibration
+    str = calibration_stage == CALIBRATION_SUCCESS ? "Calibration Completed" : "Calibration Failed, send M995 to try again";
```
Comment (thinkyhead) [inline Marlin/src/lcd/extui/lib/mks_ui/draw_touch_calibration.cpp]: For these screens we use `GET_TEXT_F` to get strings for the current language. It looks like this and some other ExtUI screens are still using fixed English strings where localized strings would be better.

**Comment by thinkyhead** [comment]
Looking pretty good. Later when this is merged I'll replace the 'compatibility' section with sanity checks (and update all the example configs).

File: Marlin/src/lcd/extui/lib/mks_ui/draw_touch_calibration.cpp (hunk 1 above)
Comment (rhapsodyv) [reply Marlin/src/lcd/extui/lib/mks_ui/draw_touch_calibration.cpp]: well.... the origin of this is: to create a generic touch calibration screen, we would have our separated font to draw the screen, and that would be limited to save flash..... but, in the middle of the dev, I changed it, and made the UI responsible to draw itself. This now allow us to use i18n here. I will update the code.

**Comment by rhapsodyv** [comment]
Done. Added i18n for touch calibration screen.

**Pull request merged**